platform-sdk/docs/base/base.md

```diff
@@ -0,0 +1,49 @@
+# Platform Base
+
+Platform base is a superset of modules that belong to the base team and are used everywhere within the project.
+
+## Goals
+
+The modules that belong to Platform Base must fulfill the following requirements:
+
+- They must provide an easy to understand API
+- They must provide documentation for all public interfaces, classes, methods...
+- They must provide documentation for each package (`package-info.java`)
+- Each module must be a Java module
+- Today all features must be supported on the modulepath and the classpath (might be modulepath only in future).
+- If the project provide test fixtures, they must be provided as a module as well.
+- A goal is to have all unit tests defined as modules, too.
```

Comment:
This should not be a goal because then packages cannot be replicated in test code, so code exposure for test purposes is greatly expanded, making encapsulation almost non-existent.
Unit tests not in modules is perfectly reasonable, particularly as unit test code should never be reused or published.

Reply:
Well but code behaves differently on the module path against the class path. Therefore having unit test per definition on the class path ends in different behavior at test time VS at runtime. Replicating packages is a pattern that allows tests to sneak into classed by accessing package private methods. From my point of view we now have a really good separation between public and private API and a private API does not need to be package private anymore. Just move it to a private package and export it only for the test.

Reply:
SPI works in a different way for example between module path and class path. Reflection is another example. The definition in the module-info of a project or of dependent project can change how code behaves. That is just a fact.

Reply:
These are still packaging and integration differences.
Sacrificing object encapsulation to try to find integration issues in unit tests is, in my opinion, a flawed choice.
Integration or cross-cut tests are a much better place to discover such things.  Those tests never need access to object internals or package private elements, absolutely should be in different modules/packages, and can discover these external integration issues more effectively than unit tests.

Reply:
I need to disagree on that one. If I have a unit that internally uses reflection I already have that problem. But that can be solved by having some definitions (for example using Google Auto Service for SPI). Anyway we need a global (platform+services) definition how tests (unit, integration,...) are executed.

Reply:
I removed that point for now to get the PR merged ;)